### PR TITLE
Pull warehouses

### PR DESCRIPTION
**Pull request by mnatalia**

### CODE REVIEW

File: opencart_api/items.py [hunk 1]
```diff
@@ -347,6 +347,12 @@ def get_item(site_name, oc_product_id):
         return frappe.get_doc('Item', db_item.get('name'))
 
 
+def get_warehouse(site_name, oc_warehouse_id):
```
Comment (ahonko) [inline opencart_api/items.py]: why is this method located in items.py?

Comment (mnatalia) [inline opencart_api/items.py]: because I used it there, use it from warehouses.py?

Comment (ahonko) [inline opencart_api/items.py]: + why do we need 2 same methods?

File: opencart_api/items.py [hunk 2]
```diff
@@ -846,6 +892,16 @@ def pull_products_from_oc(site_name, silent=False):
                             'date_end': oc_discount.get('date_end'),
                         })
 
+                    # warehouses
+                    for oc_warehouse in oc_product.get('warehouses'):
+                        doc_item.append('oc_warehouses', {
+                            'item_name': doc_item.get('name'),
+                            # 'warehouse': oc_warehouse.get('warehouse_id'),
+                            'quantity': oc_warehouse.get('quantity_product_warehouse'),
+                            'subtract_wh': oc_warehouse.get('subtract_wh'),
```
Comment (ahonko) [inline opencart_api/items.py]: if you take a look at Opencart site, this value looks differnt than in ERPnext

File: opencart_api/items.py [hunk 3]
```diff
@@ -846,6 +892,16 @@ def pull_products_from_oc(site_name, silent=False):
...
+                            'stock_status_id_wh': oc_warehouse.get('stock_status_id_wh'),
```
Comment (ahonko) [inline opencart_api/items.py]: the same as above

File: opencart_api/models.py [hunk 4]
```diff
@@ -124,6 +124,22 @@ def __repr__(self):
         return '%s - %s' % (self.id, self.name)
 
 
+class OpencartWarehouse(object):
```
Comment (ahonko) [inline opencart_api/models.py]: we don't need this class, 'cause old way that was used earlier
take a look at sales order, etc we should use directly json object that comes from Opencart API

File: opencart_api/oc_api.py [hunk 5]
```diff
@@ -291,3 +292,11 @@ def get_all_manufacturers(self, limit=100, page=1):
...
             for c in resp.get('data', []):
                 yield c
+
+    def get_warehouses(self):
+        success, resp = oc_request('http://store.fortafy.com/index.php?route=rest' + '/product_admin/warehouses', headers=self.headers)
+        if not success:
+            return
+        warehouses = resp.get('data', [])
+        for i in warehouses:
+            yield OpencartWarehouse(self, i)
```
Comment (ahonko) [inline opencart_api/oc_api.py]: as above mentioned, we don't use this old-fashion classes from from models.py

**Pull request closed without merging**